Remove legacy hash, cipher, cmac, aead and rsa crypto options

ronald-cron-arm opened the pull request:
## Description
Remove legacy hash, cipher, cmac, aead and rsa crypto options
Progresses #360, #361 and #363.

## PR checklist
- [x] **changelog** not required because: will be done for all legacy crypto options as part of #https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/364
- [x] **framework PR** provided Mbed-TLS/mbedtls-framework#183, preparatory work to get a green CI here
- [x] **mbedtls development PR** provided Mbed-TLS/mbedtls#10297, preparatory work to get a green CI here
- [x] **mbedtls 3.6 PR** not required because: Legacy options will not be removed in 3.6
- **tests**  not required because: removing some configuration options, we just maintain the test coverage.

ronald-cron-arm commented:
Only one CI glitch (Session resume using cache, DTLS: openssl server ....................... RETRY(client-timeout) RETRY(client-timeout) FAIL), the CI can be considered as green.

davidhorstmann-arm reviewed (dismissed):
LGTM, thanks!

minosgalanakis commented on include/psa/crypto_config.h:
```diff
@@ -2101,35 +1995,6 @@
- */
-#define MBEDTLS_CIPHER_PADDING_PKCS7
-#define MBEDTLS_CIPHER_PADDING_ONE_AND_ZEROS
-#define MBEDTLS_CIPHER_PADDING_ZEROS_AND_LEN
```
`MBEDTLS_CIPHER_PADDING_ZEROS_AND_LEN` and `MBEDTLS_CIPHER_PADDING_ZEROS` have been removed but there is still code using them in drivers -> [cipher.h ](https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/development/drivers/builtin/include/mbedtls/cipher.h#L196)
and [tests/suites/test_suite_cipher.function](https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/development/tests/suites/test_suite_cipher.function#L176)

Is that intended to be cleaned up in a future pr?

minosgalanakis reviewed (commented):
Looks good overall and clears CI, but there are some minor questions I would like to be clarify before approving

ronald-cron-arm commented on drivers/builtin/src/asn1write.c:
```diff
@@ -456,7 +456,9 @@ int mbedtls_asn1_write_integer(unsigned char **p,
 
-    integer_start = integer + number_of_leading_zeros;
+    if (integer != NULL) {
+        integer_start = integer + number_of_leading_zeros;
```
This function is actually new, see `ChangeLog.d/asn1-write-integer.txt` thus no need for a change log.

ronald-cron-arm commented:
I've rebased due to conflicts with `framework` and `crypto_config.h`. With the rebase the commit updating the framework pointer has disappeared. The conflict in `crypto_config.h` was due to the removal of the MBEDTLS_HKDF_C option.
Otherwise, I've added an additional commit to address @minosgalanakis [comment](https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/379#discussion_r2219386691).

ronald-cron-arm commented:
The PR was approved by @davidhorstmann-arm at https://github.com/Mbed-TLS/TF-PSA-Crypto/commit/87ab5a14dcfdd125f210d543e1e528b45ce713af. @bensze01 please review the rebase and the additional last commit, thanks.

minosgalanakis reviewed (approved):
LGTM

ronald-cron-arm commented:
https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/379/commits/6cde841bcc8a310c813afb152b81278ca732a84a has been approved by the CI and is based on tf-psa-crypto/development thus merging directly.
